server/src/updaters/relationship-updaters.js

```diff
@@ -199,7 +201,65 @@ async function updateRelationships(
 }
 
 function sortIDs(firstId: string, secondId: string) {
-  return [firstId, secondId].map(Number).sort((a, b) => a - b);
+  return [Number(firstId), Number(secondId)].sort((a, b) => a - b);
 }
 
-export { updateRelationships };
+async function knowOfRelationshipsForThread(
+  newMemberIDs: string[],
+  threadID?: string,
+): Promise<RelationshipChangeset> {
+  const combinations = combinations2(newMemberIDs).map(
+    ([firstId, secondId]) => {
+      const [user1, user2] = sortIDs(firstId, secondId);
+      return { user1, user2, status: undirectedStatus.KNOW_OF };
+    },
+  );
+
+  if (threadID) {
+    const query = SQL`
+      SELECT user FROM memberships WHERE thread = ${threadID} ORDER BY user
+    `;
+    const [result] = await dbQuery(query);
+    const memberIDs: number[] = result.map(membership => membership.user);
+
+    return newMemberIDs
+      .reduce((acc, newMemberID) => {
+        return acc.concat(
+          memberIDs.map(memberID => {
+            const [user1, user2] = sortIDs(memberID.toString(), newMemberID);
+            return { user1, user2, status: undirectedStatus.KNOW_OF };
+          }),
+        );
+      }, [])
+      .concat(combinations);
+  } else {
+    return combinations;
+  }
+}
+
+async function updateUndirectedRelationships(
+  changeset: RelationshipChangeset,
+  override: boolean = true,
+) {
+  if (!changeset.length) {
+    return;
+  }
+
+  const rows = changeset.map(row => [row.user1, row.user2, row.status]);
+  const query = SQL`
+    INSERT INTO relationships_undirected (user1, user2, status)
+    VALUES ${rows}
+  `;
+  if (override) {
+    query.append('ON DUPLICATE KEY UPDATE status = VALUES(status)');
```

Comment:
Nit: let's use SQL template literals here for both of the `query.append` calls. It's true we don't need them here because we don't have any inline variables, but I think it's a good practice to always use the SQL template literals. It makes it clear to the reader that you're writing SQL, and it makes sure that if somebody later adds an inline variable, it will be escaped properly.